# HG changeset patch
# User Thorsten Liebig <thorsten.liebig@gmx.de>
# Date 1346959264 -7200
#      Thu Sep 06 21:21:04 2012 +0200
# Node ID 5ddeef055df32175cb5a211197ceb60129c2bddf
# Parent  eefbbc1ed60ed0c0c7c60e429ea72dc473fb7bb9
* file-editor.cc (file_editor::request_open_file): canceled open --> restore focus to previous tab

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -94,28 +94,32 @@ file_editor::request_new_file ()
       fileEditorTab->new_file ();
     }
 }
 
 void
 file_editor::request_open_file ()
 {
   file_editor_tab *current_tab = active_editor_tab ();
+  int curr_tab_index = _tab_widget->currentIndex ();
   file_editor_tab *fileEditorTab = new file_editor_tab (this);
   if (fileEditorTab)
     {
       add_file_editor_tab (fileEditorTab);
       QString dir = QDir::currentPath ();
       // get the filename of the last active tab to open a new file from there
       if (current_tab)
         dir = QDir::cleanPath (current_tab->get_file_name ());
       if (!fileEditorTab->open_file (dir))
         {
           // If no file was loaded, remove the tab again.
           _tab_widget->removeTab (_tab_widget->indexOf (fileEditorTab));
+          // restore focus to previous tab
+          if (curr_tab_index>=0)
+            _tab_widget->setCurrentIndex (curr_tab_index);
         }
     }
 }
 
 void
 file_editor::request_open_file (const QString& fileName)
 {
   if (!isVisible ())
